### ISSUE TITLE
[core] Automatic cherry-pick to the `master` branch

### ISSUE THREAD

**Issue by cherniavskii** [mui/mui-x#11272]
The flow:
- Add a `needs cherry-pick` label to the PR if it needs to be cherry-picked to the `master` branch
- Once the PR is merged into the `next` branch, the merge commit gets cherry-picked to the new branch started of the `master` branch, and a new PR is opened targeting the `master` branch

**Search keywords**:

**Comment by cherniavskii** [comment]
Closed by https://github.com/mui/mui-x/pull/11382
